idr/marshal1.go

```diff
@@ -0,0 +1,54 @@
+package idr
+
+import (
+	"fmt"
+
+	"github.com/jf-tech/go-corelib/jsons"
+)
+
+// j1NodePtrName returns a categorized name for a *Node pointer used in JSONify1
+func j1NodePtrName(n *Node) string {
+	if n == nil {
+		return "(nil)"
+	}
+	name := func(n *Node) string {
+		if IsXML(n) && XMLSpecificOf(n).NamespacePrefix != "" {
```

Comment:
Actually I am not sure which one is better: convert to `XMLSpecific` to check type, or specify one `type` in FormatSpecific as `xml` so that may be easier to use switch later?
BTW, shall we use some string not empty function consider trim situation or directly compared with `""` here?

Reply:
Just don't want to introduce any other field that takes memory for no reason. If the type of `FormatSpecific` can do the job, no need for an extra `type` field.

I also start to be more cautious of over-generous use of strings.TrimSpace which takes time and lots of situations make no difference. Here, 1) `NamespacePrefix` for all practically purposes are populated our xml reader so it's considered internal, so need for trimming and 2) even with leading/trailing spaces, an URI is a URI, we don't trim URI.